Array_Functions/Binary_Search/binary_search.cpp

```diff
@@ -0,0 +1,51 @@
+//Implementatiom of Binary Search Algorithm in C++
+#include <bits/stdc++.h>
+using namespace std;
+
+int bsearch(double arr[], int n, double x)
+{
+    int mid, low = 0, high = n - 1;
+    while (low <= high)
+    {
+        mid = (low + high) / 2;
```

Comment:
Good, your solution is correct. Only a slight modification is required, to handle cases when 'n' is significantly large. In that case if you calculate mid = (low + high)/2, then the value may overflow. Instead, use mid = (high - low)/2 + low. This is the most standard practice which is used.

Modify this in all three files

Reply:
Yeah I saw that somewhere and it did not make sense to me at that time. Thanks, will make the necessary changes.